packages/block-editor/src/store/utils.js

```diff
@@ -141,5 +144,6 @@ export const getInsertBlockTypeDependants =
 			state.blockEditingModes,
 			select( STORE_NAME ).__unstableGetEditorMode( state ),
 			getSectionRootClientId( state ),
+			getParentSectionBlock( state, rootClientId ),
```

Comment:
It makes sense that it's a memoization issue, though I can't see where `getParentSectionBlock` is used in `canInsertBlockTypeUnmemoized`.

I do see `isSectionBlock`, so I wondered if that should be present ... and also whether it solves the bugs if it's in place of `getParentSectionBlock`.